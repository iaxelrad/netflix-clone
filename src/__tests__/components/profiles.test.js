import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../components';

describe('Profiles', () => {
  it('renders <Profiles /> with populated data', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture
              src="/images/test.png"
              data-testid="profile-picture"
            />
            <Profiles.Name>Test Name</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId('profile-picture')).toBeTruthy();
    expect(getByText('Test Name')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders <Profiles /> with populated data but misc profile image', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture data-testid="profile-picture-misc" />
            <Profiles.Name>Test Name</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId('profile-picture-misc')).toBeTruthy();
    expect(getByText('Test Name')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
