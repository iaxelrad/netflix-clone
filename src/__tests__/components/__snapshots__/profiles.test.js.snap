// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profiles renders <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-dt3hl3-0 cDXJlL"
>
  <h1
    class="profiles__Title-dt3hl3-1 jUdiKh"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-dt3hl3-2 elHKjw"
  >
    <li
      class="profiles__Item-dt3hl3-5 TEflU"
    >
      <img
        class="profiles__Picture-dt3hl3-3 dvUhoM"
        data-testid="profile-picture"
        src="/images/users//images/test.png.png"
      />
      <p
        class="profiles__Name-dt3hl3-4 lolVpr"
      >
        Test Name
      </p>
    </li>
  </ul>
</div>
`;

exports[`Profiles renders <Profiles /> with populated data but misc profile image 1`] = `
<div
  class="profiles__Container-dt3hl3-0 cDXJlL"
>
  <h1
    class="profiles__Title-dt3hl3-1 jUdiKh"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-dt3hl3-2 elHKjw"
  >
    <li
      class="profiles__Item-dt3hl3-5 TEflU"
    >
      <img
        class="profiles__Picture-dt3hl3-3 dvUhoM"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-dt3hl3-4 lolVpr"
      >
        Test Name
      </p>
    </li>
  </ul>
</div>
`;
